libs/src/services/creatorNode/CreatorNodeSelection.ts

```diff
@@ -1,24 +1,90 @@
-const _ = require('lodash')
-
-const { ServiceSelection } = require('../../service-selection/ServiceSelection')
-const { timeRequests, sortServiceTimings } = require('../../utils/network')
-const { CREATOR_NODE_SERVICE_NAME, DECISION_TREE_STATE } = require('./constants')
+import type { AxiosResponse } from 'axios'
+import _ from 'lodash'
+
+import { ServiceSelection } from '../../service-selection'
+import {
+  timeRequests,
+  sortServiceTimings,
+  ServiceWithEndpoint,
+  Service,
+  ServiceName,
+  Timing
+} from '../../utils'
+import { CREATOR_NODE_SERVICE_NAME, DECISION_TREE_STATE } from './constants'
+
+type Timeout = number | null
 
 /**
  * In memory dictionary used to query spID from endpoint
  * Eliminates duplicate web3 calls within same session
  */
-const contentNodeEndpointToSpID = { }
-function getSpIDForEndpoint (endpoint) {
+const contentNodeEndpointToSpID: Record<string, unknown> = {}
+
+export function getSpIDForEndpoint(endpoint: string) {
   return contentNodeEndpointToSpID[endpoint]
 }
 
-function setSpIDForEndpoint (endpoint, spID) {
+export function setSpIDForEndpoint(endpoint: string, spID?: string) {
   contentNodeEndpointToSpID[endpoint] = spID
 }
 
-class CreatorNodeSelection extends ServiceSelection {
-  constructor ({
+type CreatorNode = {
+  getSyncStatus: (
+    service: Service,
+    timeout: Timeout
+  ) => Promise<{ isBehind: boolean; isConfigured: boolean }>
+  passList?: Set<string>
+  blockList?: Set<string>
+  monitoringCallbacks: {
+    healthCheck?: (config: Record<string, unknown>) => Promise<AxiosResponse>
+  }
+}
+
+type EthContracts = {
+  getCurrentVersion: (serviceName: string) => Promise<string>
+  getNumberOfVersions: (spType: string) => Promise<number>
+  getVersion: (spType: string, queryIndex: number) => Promise<string>
+  getServiceProviderList: (
+    serviceName: string
+  ) => Promise<ServiceWithEndpoint[]>
+  hasSameMajorAndMinorVersion: (version1: string, version2: string) => boolean
+  isInRegressedMode: () => boolean
+}
+
+type CreatorNodeSelectionConfig = {
```

Comment:
Could this be `ServiceSelectionConfig & `{ ... }`?

Reply:
yeah great idea!